src/BaGet.Core.Server/Controllers/PackageContentController.cs

```diff
@@ -79,5 +79,21 @@ public async Task<IActionResult> DownloadReadmeAsync(string id, string version,
 
             return File(readmeStream, "text/markdown");
         }
+
+        public async Task<IActionResult> DownloadIconAsync(string id, string version, CancellationToken cancellationToken)
+        {
+            if (!NuGetVersion.TryParse(version, out var nugetVersion))
+            {
+                return NotFound();
+            }
+
+            var iconStream = await _content.GetPackageIconStreamOrNullAsync(id, nugetVersion, cancellationToken);
+            if (iconStream == null)
+            {
+                return NotFound();
+            }
+
+            return File(iconStream, "image/png");
```

Comment:
The content type `image/png` may be incorrect. Let's copy nuget.org's logic: https://github.com/NuGet/NuGet.Services.Metadata/blob/91f97be8882ec83c3dfd7521ebe62865ab3adb5d/src/Catalog/Icons/IconProcessor.cs#L170-L278

Reply:
Some notes:

* I would add a `DetermineContentTypeAsync` to BaGet's [`StreamExtensions`](https://github.com/loic-sharma/BaGet/blob/05bfc61e1ff86517d2e4571f12524275c2355a0d/src/BaGet.Core/Extensions/StreamExtensions.cs). That way you can do:

```csharp
var contentType = iconStream.DetermineContentTypeAsync(cancellationToken);
return File(iconStream, contentType);
```

* When copying the [`DetermineContentType`](https://github.com/NuGet/NuGet.Services.Metadata/blob/91f97be8882ec83c3dfd7521ebe62865ab3adb5d/src/Catalog/Icons/IconProcessor.cs#L227-L230) method from NuGet.org, you can act as if `onlyGallerySupported` is `true`:

```csharp
// Checks are ordered by format popularity among external icons for existing packages
if (ArrayStartsWith(imageData, PngHeader))
{
    return "image/png";
}

if (ArrayStartsWith(imageData, JpegHeader))
{
    return "image/jpeg";
}

return "";
```

Reply:
It turns out using an empty string `""` does not work. I will try `image/xyz` instead.